Formula/reprepro.rb

```diff
@@ -0,0 +1,37 @@
+class Reprepro < Formula
+  desc "Debian package repository manager"
+  homepage "https://salsa.debian.org/brlink/reprepro"
+
+  url "https://deb.debian.org/debian/pool/main/r/reprepro/reprepro_5.3.0.orig.tar.gz"
+  sha256 "5a5404114b43a2d4ca1f8960228b1db32c41fb55de1996f62bc1b36001f3fab4"
+
+  depends_on "berkeley-db@4"
+  depends_on "gpgme"
+  depends_on "libarchive"
+  depends_on "xz"
+
+  fails_with :clang do
+    cause "No support for GNU C nested functions"
```

Comment:
Yuck, installing GCC as hard dependency for such a simply software seems very heavy :(

Reply:
Yeah, it's unfortunate that upstream uses certain GNU C extensions that aren't implemented in other compilers. :(

Thankfully it's just a build-time dependency, so it won't affect most people installing the formula. The heavy build-time dependencies are one of my main motivations for packaging this for Homebrew.

Reply:
GCC is not a build dependency, it's a runtime dep:

```
$ otool -L /usr/local/Cellar/reprepro/5.3.0/bin/reprepro | grep gcc
	/usr/local/lib/gcc/8/libgcc_s.1.dylib (compatibility version 1.0.0, current version 1.0.0)
```

since code compiled with GCC has to be linked against its runtime library.

Please add `depends_on "gcc"` to make it clearer.

Reply:
Wow, TIL about `libgcc`. Thanks for catching this @fxcoudert, I've added `depends_on "gcc"` here.

I reviewed existing homebrew-core formulas that require GCC for compilation, and it seems like simply adding a runtime `gcc` dependency is indeed the current convention.

I read about libgcc in the GCC manual (https://gcc.gnu.org/onlinedocs/gccint/Libgcc.html), and learned that it is very lightweight, and primarily provides implementations of mathematical operations that are not supported on a target architecture's hardware plus some other C++ exception-handling functionality.

I think an alternative to adding a hard runtime dependency on `gcc` in many Homebrew packages such as this one would be to add `-static-libgcc` to `LDFLAGS`.

Static linking libgcc could be appropriate because the library is extremely lightweight, and in some cases may even be strictly better in terms of simplifying generated binaries: for example, for this formula, I tested compiling with and without `-static-libgcc` and there didn't seem to be any effect on compiled code size. In fact, even though `/usr/local/lib/gcc/8/libgcc_s.1.dylib` is listed as a dylib dependency, `nm -m` didn't list any symbols to be imported from this dylib at all at runtime!

Thoughts / should I ticket this as a possible change in Homebrew itself?

Reply:
Static libgcc increases the size of all binaries and libraries, so it needs to be a case-by-case decision, not something that Homebrew does automatically behind your back.

But if the software can detect and build with a static libgcc (say, with a configure flag) then yes, in that case we could have `gcc` as a build dependency, which would be ideal for users.